doc/sphinx/source/developer_guide/logging.rst

```diff
@@ -0,0 +1,264 @@
+.. developer_guide logging
+
+=============
+Logging
+=============
+
+The ability to log and record drone data is a crucial element in a drone's mission as well
+as for debugging purposes. There are different methods for data logging, both over telemetry and
+onboard on an external device, all of which are covered in this section.
+
+
+Logging on Linux based OS
+------------------------------
+The approaches discussed in this section are only possible for systems that run a Linux based OS
+with access to internal storage space on your autopilot, as is the case with most Parrot drones.
+
+Printing to terminal
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Perhaps one of the easiest approaches to logging is to make your module print to a terminal.
+You can add your own debug messages by including the ``<stdio.h>`` header to get access to the
+``fprintf`` function. Then, at any point in your code, you can use this function to print your own
+message.
+
+.. code-block:: C
+
+  <stdio.h>
+  fprintf(stderr, "<Your debug message>");
+
+It can be useful (if you expect to use this method a lot) to create a macro for this command that
+automatically includes additional information on where the message originates, like is done in the
+``orange_avoider`` module.
+
+.. code-block:: C
+
+  #define PRINT(string,...) fprintf(stderr, "[orange_avoider->%s()] " string,__FUNCTION__ , ##__VA_ARGS__)
+
+This allows you to only have to write ``PRINT("<your debug message>")`` in your code and it will automatically
+prepend ``"[orange_avoider->%s()] "`` to your message, which can help debugging if multiple modules
+are printing information simultaneously.
+
+When running in simulation ``fprintf`` automatically prints the message in the Paparazzi Center. You can also
+read these messages on the real drone using the following steps:
```

Comment:
I suggest to specify that this example is only valid for Bebop 1 et 2.

Reply:
Is this not possible with every Linux based OS that has WiFi and specifies a host for a telnet connection? E.g. the Disco, although during flight you wouldn't be able to maintain WiFi connection

Reply:
It's true that it works for Disco which is sharing the same hardware than Bebop more or less. The only Linux-based system that we support are Parrot drones, so I guess it is fine even if there are so things specific to these platforms, like default IP or directory tree.